modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/BloggingPageHelper.cs

```diff
@@ -102,39 +104,36 @@ public LocalizedHtmlString ConvertDatetimeToTimeAgo(DateTime dt)
 
             var diffInDays = (int) timeDiff.TotalDays;
 
-            if (diffInDays >= 365)
+            switch (diffInDays)
             {
-                return  L["YearsAgo", diffInDays / 365];
-            }
-            if (diffInDays >= 30)
-            {
-                return L["MonthsAgo", diffInDays / 30];
-            }
-            if (diffInDays >= 7)
-            {
-                return L["WeeksAgo", diffInDays / 7];
-            }
-            if (diffInDays >= 1)
-            {
-                return L["DaysAgo", diffInDays];
+                case >= 365:
+                    return ConvertDatetimeToTimeAgo("YearsAgo", "YearAgo", diffInDays / 365);
+                case >= 30:
+                    return ConvertDatetimeToTimeAgo("MonthsAgo", "MonthAgo", diffInDays / 30);
+                case >= 7:
+                    return ConvertDatetimeToTimeAgo("WeeksAgo", "WeekAgo", diffInDays / 7);
+                case >= 1:
+                    return ConvertDatetimeToTimeAgo("DaysAgo", "DayAgo", diffInDays);
             }
 
             var diffInSeconds = (int) timeDiff.TotalSeconds;
 
-            if (diffInSeconds >= 3600)
+            switch (diffInSeconds)
             {
-                return L["HoursAgo", diffInSeconds / 3600];
+                case >= 3600:
+                    return ConvertDatetimeToTimeAgo("HoursAgo", "HourAgo", diffInSeconds / 3600);
+                case >= 60:
+                    return ConvertDatetimeToTimeAgo("MinutesAgo", "MinuteAgo", diffInSeconds / 60);
+                case >= 1:
+                    return  ConvertDatetimeToTimeAgo("SecondsAgo", "SecondAgo", diffInSeconds);
+                default:
+                    return L["Now"];
             }
-            if (diffInSeconds >= 60)
-            {
-                return L["MinutesAgo", diffInSeconds / 60];
-            }
-            if (diffInSeconds >= 1)
-            {
-                return  L["SecondsAgo", diffInSeconds];
-            }
-
-            return L["Now"];
+        }
+        
+        protected virtual LocalizedHtmlString ConvertDatetimeToTimeAgo(string pluralKey, string singularKey, int value)
```

Comment:
Please rename this method's name. It might be something like *GetLocalizedTimeAgoText*. Because currently, it indicates that it casting from one type to another one because of the method name being started with *Convert...*